lib/internal/perf/timerify.js

```diff
@@ -80,8 +81,8 @@ function timerify(fn, options = {}) {
     const result = constructor ?
       ReflectConstruct(fn, args, fn) :
       ReflectApply(fn, this, args);
-    if (!constructor && typeof result?.finally === 'function') {
-      return result.finally(
+    if (!constructor && typeof result?.then === 'function') {
+      return result.then(
         FunctionPrototypeBind(
```

Comment:
not like `bind` here. I feel `v => { processComplete(fn.name, start, args, histogram); return v }` would be much clear.